src/sound/clip.cpp

```diff
@@ -0,0 +1,159 @@
+#include "clip.hpp"
+
+SoundClip::SoundClip(){
+	alGetError();
+	alGenBuffers(1, &buffer);
+	
+	ALenum error;
+	if ((error = alGetError()) != AL_NO_ERROR){
+		std::cerr << ("Clip -> SCinit: Failed to generate buffers\n");  //throw exception
+		return;
+	}
+}
+
+SoundClip::~SoundClip(){
+	alDeleteBuffers(1, &buffer);
+}
+
+void SoundClip::loadOGGFile(const char* filename) {
+	// clean up errors
+	alGetError();
+
+	int channels, sample_rate;
+	short* output;
+
+	// get output, channels, sample rate from filename and size of the data
+	ALsizei size_block = stb_vorbis_decode_filename(filename, &channels, &sample_rate, &output);

```

Comment:
The memory allocated by this method is incorrectly freed, 
STB Vorbis is a C library and uses `malloc()` buffers allocated by it must be freed with `free()` not `delete[]`
For a easy fix repalce all usages of C++ `new/delete` with `malloc/free` here